Makefile

```diff
@@ -46,18 +46,18 @@ endif
 #
 
 smoke-test:	test-clean-up	build_test
-	$(TEST_BIN) smoke --spu ${DEFAULT_SPU} --produce-iteration ${DEFAULT_ITERATION} --local ${TEST_LOG} ${SKIP_CHECK} ${EXTRA_ARG}
+	$(TEST_BIN) --spu ${DEFAULT_SPU} --local ${TEST_LOG} ${SKIP_CHECK} ${EXTRA_ARG} -- smoke --producer-iteration=${DEFAULT_ITERATION}
```

Comment:
Hmm.   This is little bit awkward.  I think we should still keep test name as argument before so: it should be like:
```
$(TEST_BIN) smoke  --spu ${DEFAULT_SPU} --local ${TEST_LOG} ${SKIP_CHECK} ${EXTRA_ARG} -- producer-iteration=${DEFAULT_ITERATION}
```

You can still inject test name by manipulating command